util/testutil/directory.go

```diff
@@ -84,15 +87,20 @@ func NewCallbackCloser(fn func()) Closer {
 }
 
 func (t temporaryDirectory) Close() {
+	retries := temporaryDirectoryRemoveRetries
```

Comment:
Yes, it helps. In a Windows machine, the promql tests always fail due to **RemoveAll()** can't remove a directory because is not empty. The strange thing is that if you go to the offending directory after the tests finished, the directory is actually empty.
My first approach to fix this, was to check if the Storage was properly close before trying to delete the storage directory, but I couldn't find any issue in that (all the proper channels seems to be wait before returning the control to the calling function). Then I checked, if was LevelDB, but I didn't have any luck founding the issue. So, maybe this is a bug of RemoveAll() in Windows, maybe Prometheus is giving back the control before all the things are properly close or maybe LevelDB is the one that is giving back the control before it should.